php/php.editor/tools/ASTPHP5Parser.cup

```diff
@@ -5360,23 +5435,13 @@ T_OPEN_RECT:o dim_offset:index T_CLOSE_RECT:end
 {:
     RESULT = new ArrayDimension(oleft, endright, index, ArrayDimension.Type.VARIABLE_ARRAY);
 :}
-
-| T_CURLY_OPEN:o dim_offset:index T_CURLY_CLOSE:end
-{:
-    RESULT = new ArrayDimension(oleft, endright, index, ArrayDimension.Type.VARIABLE_HASHTABLE);
-:}
```

Comment:
Note: Curly braces array access (`{}` e.g. `$array{1}`, `$array{'key'}`) can use no longer.
Because a conflict occurs in the following case:

```php
"string"{1};
public string $prop = "string" {
    get => $this->prop;
    set {}
}
```

Reply:
This should be fine, hopefully.

Reply:
Yes :)
- deprecated as of PHP 7.4
- fatal error as of PHP 8.0
- parse error as of PHP 8.4